From 071c0d19396be0a12d9e2300d6fa2667ad6f032e Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 5 Sep 2011 17:36:06 +0200
Subject: [PATCH 2/6] Xi: Ensure touch events update currentTime

This is so grabs initiated by touch events don't check
against a possibly outdated currentTime, and appear to
be in the future while they actually aren't.
---
 Xi/exevents.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Xi/exevents.c b/Xi/exevents.c
index b8b9129..d445c9a 100644
--- a/Xi/exevents.c
+++ b/Xi/exevents.c
@@ -1924,6 +1924,7 @@ ProcessOtherEvent(InternalEvent *ev, DeviceIntPtr device)
     DeviceEvent *event = &ev->device_event;
 
     CHECKEVENT(ev);
+    UpdateCurrentTimeIf();
 
     if (ev->any.type == ET_RawKeyPress ||
         ev->any.type == ET_RawKeyRelease ||
-- 
1.7.5.4

